src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -121,4 +124,57 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
         }
         return tagSet;
     }
+
+    /**
+     * Parses a {@code String dateTime} into an {@code LocalDateTime}.
+     *
+     * @throws ParseException if the given {@code dateTime} is invalid.
+     */
+    public static LocalDateTime parseDateTime(String dateTime) throws ParseException {
+        requireNonNull(dateTime);
+        String trimmedDateTime = dateTime.trim();
+
+        DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        LocalDateTime localDateTime;
+        try {
+            localDateTime = LocalDateTime.parse(trimmedDateTime, format);
+        } catch (Exception e) {
+            throw new ParseException(Appointment.MESSAGE_CONSTRAINTS);
+        }
+        return localDateTime;
+    }
+
+    /**
+     * Parses a {@code Collection<String> attend} into a {@code boolean}.
+     * By default, false will be returned if {@code attend} is empty.
+     *
+     * @throws ParseException if the given {@code attend} is invalid.
+     */
+    public static boolean parseHasAttended(Collection<String> attend) throws ParseException {
+        requireNonNull(attend);
+        if (attend.isEmpty()) {
+            return false;
+        }
+
+        String trimmedAttend = attend.iterator().next().trim();
+        if (!trimmedAttend.equalsIgnoreCase("true") &&
+                !trimmedAttend.equalsIgnoreCase("false")) {
+            throw new ParseException(Appointment.MESSAGE_CONSTRAINTS);
+        }
+        return Boolean.parseBoolean(trimmedAttend);
+    }
+
+    /**
+     * Parses a {@code Collection<String> description} into a {@code String}.
+     * Leading and trailing whitespaces will be trimmed.
+     * By default, an empty string will be returned if {@code description} is empty.
+     * TODO: remove after case log is implemented
+     */
+    public static String parseDescription(Collection<String> description) {
```

Comment:
nit: Why not use List<String> instead of Collection?

Reply:
I was follwing the implementation for parseTags. Using List would work as well. Is using List better than Collection?

Reply:
Nope there is no difference. But then now I have another question, why use a Collection of Strings instead of just a String, like Person Name?

Reply:
Sorry for the late reply. This is because hasAttended and appointmentDescription are optional parameters. 

When parsing the input command, `argMultimap.getValue(PREFIX_APPOINTMENT_DESCRIPTION).get()` will have a `java.lang.reflect.InvocationTargetException` error if the appointment description is not present in the command (`eg command: appointment sid/1 d/2024-03-18 09:00`). Therefore, I am not able to use the `getValue` function to retrieve and pass a string in to the respective parser function.

However if I use `argMultimap.getAllValues(PREFIX_APPOINTMENT_DESCRIPTION)` it will collect all fields with the prefix `ad` and will cause an error event if there is no appointment description provided. The `getAllValues` function returns `Collection<String>`.

Reply:
@caitlyntang I think tis' still possible. Can you try using `argMultimap.getValue(PREFIX_APPOINTMENT_DESCRIPTION).orElse("")`?

Reply:
You are right, `argMultimap.getValue(PREFIX_APPOINTMENT_DESCRIPTION).orElse("")` resolves the error. Will make the appropriate changes!